### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by sainingo**
### Interactions
**Splash screen**

A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
Sign up and log in pages

- The user should be able to register in the app with full name, email and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (catagories page)

When the user logs in, they are presented with the categories page.

- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page

For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).

- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page
- 

The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new transaction" page

The user fills out a form to create a new transaction with:

- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)
- The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: app/views/categorys/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ <img class="w-[80%] h-[40%]" src="<%= category.icon%>" alt="...">
+ <h1 class="text-2xl font-bold m-4"><%= link_to category.name, category_path(category.id)%></h1>
+  <%= link_to "View",  category_path(category.id), class: "bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"%>
+  </div>
+<% end%>
+</div>
+
+<div class="mt-2">
+<%= link_to "Add New Category", new_category_path, class: "bg-blue-500 hover:bg-blue-700 text-white font-bold py-3 px-4 rounded ml-8"%>
+</div>
```
Comment (thecodechaser) [inline app/views/categorys/index.html.erb]: - [ ] Please also display the total amount of transactions belonging to each category.
image

image

File: app/views/transactions/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+</button>
+  <% @transactions.each do |transaction| %>
+    <%= render transaction %>
+    <p class="m-4">
+      <%= link_to "Show this transaction", category_transaction_path(id: transaction.id), class: "bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded m-8" %>
+    </p>
+  <% end %>
+</div>
+
+<%= link_to "New transaction", new_category_transaction_path, class: "bg-green-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded mt-4" %>
```
Comment (thecodechaser) [inline app/views/transactions/index.html.erb:16]: - [x] I can't see the transactions list on the transaction page. Please make sure to display all transactions belonging to that category on the transaction page ordered by the most recent.
image

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @sainingo👋, 

### You are doing a marvelous job on the RoR capstone project and this is one of the best projects I have seen so far!

There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- All tests are passed. ✔️



## Required Changes ♻️

- [x] When I run the application I see the log-in/and signup pages. Please create a splash page(Welcome page) with login and signup page links. This splash page should open when users run the application.
image

- [x] In the README file acknowledgment section, Please give appropriate credit to the author `[Gregoire Vella on Behance](https://www.behance.net/gregoirevella).` for the design.
- [x] Please make sure the video presentation of the project includes some highlighted interesting piece of code or something you built that you are very proud of.
- [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Kantush, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**